pkg/component/component.go

```diff
@@ -42,6 +42,15 @@ func GetComponentTypeFromDevfileMetadata(metadata devfile.DevfileMetadata) strin
 	return componentType
 }
 
+// GetComponentRuntimeFromDevfileMetadata returns the Language defined in the Devfile metadata
+// or if not set, the Project Type
+func GetComponentRuntimeFromDevfileMetadata(metadata devfile.DevfileMetadata) string {
+	if metadata.Language != "" {
+		return metadata.Language
+	}
+	return metadata.ProjectType
```

Comment:
I know we already discussed about using the language first and then the project type, but after quickly testing on Dev Sandbox, I think we should use the project type first.

![image](https://user-images.githubusercontent.com/593208/189861249-2fee3a13-ac27-4a7f-ac30-8f48382bac0b.png)

The Deployment I created manually in Dev Sandbox has the following label: `app.openshift.io/runtime=nodejs`, but the one created by `odo` has `app.openshift.io/runtime=javascript`, and the right icon is not showing up in Topology View.

Reply:
What is the behavioue with Java projects?

Reply:
With a Quarkus app, we have the right icon for the language (Java):

![image](https://user-images.githubusercontent.com/593208/189867907-cd8d79e0-9143-414e-82ad-57379261d1cc.png)

If I manually edit the label to use the project type (`quarkus`) defined in the Devfile, we have a more accurate icon for this:

![image](https://user-images.githubusercontent.com/593208/189868186-e13f54fb-396a-4ee3-8dd3-71bd0be71d95.png)

Reply:
Same point for a SpringBoot app: we currently have the same icon as Quarkus for the language (Java). But if I set the label to the project type (`spring`) defined in the Devfile, then we have a more specific icon for this component:

![image](https://user-images.githubusercontent.com/593208/189869705-f080fde4-6499-4ac2-96fb-da69ff288d29.png)

This is why I was thinking about using the project type first instead.

Reply:
I guess we never will be able to reliably cover all the cases. If we use projectType then there will be a problem with `java-maven`. I guess we just need to check what approach works for most of the cases